bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Runtime.java

```diff
@@ -61,6 +62,39 @@ public static Runtime getCurrentRuntime() {
         return new Runtime(strand.scheduler);
     }
 
+    /**
+     * Invoke Object method asynchronously. This will schedule the function and block the strand.
+     *
+     * @param object     Object Value.
+     * @param methodName Name of the method.
+     * @param strandName Name for newly creating strand which is used to execute the function pointer. This is
+     *                   optional and can be null.
+     * @param metadata   Meta data of new strand.
+     * @param callback   Callback which will get notify once method execution done.
+     * @param callConcurrently Ensures method can be invoked concurrently.
+     * @param properties Set of properties for strand
+     * @param returnType Expected return type of this method
+     * @param args       Ballerina function arguments.
+     * @return           {@link FutureValue} containing return value of executing this method.
+     */
+    public BFuture invokeMethodAsync(BObject object, String methodName, String strandName, StrandMetadata metadata,
+                                     boolean callConcurrently, Callback callback, Map<String, Object> properties,
+                                     Type returnType, Object... args) {
+        if (object == null) {
+            throw ErrorCreator.createError(StringUtils.fromString("object cannot be null"));
+        }
+        if (methodName == null) {
+            throw ErrorCreator.createError(StringUtils.fromString("method name cannot be null"));
+        }
```

Comment:
Can't we have a generic exceptions for all null pointers? Or since this is a java null can we throw the java `NullPointerException` or more correct to throw `IlegalArgumentException`? Saying `object cannot be null` in a Ballerina error is strange because we only have `nils` in Ballerina

Reply:
since this java api I'll change to IlegalArgumentException

Reply:
created https://github.com/ballerina-platform/ballerina-lang/issues/32225 recheck on this.